File: apollo/src/main/java/com/lsxiao.apllo/annotations/SubscribeOn.java
package com.lsxiao.apllo.annotations;

import com.lsxiao.apllo.entity.SchedulerProvider;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * write with Apollo
 * author:lsxiao
 * date:2017-04-22 02:21
 * github:https://github.com/lsxiao
 * zhihu:https://zhihu.com/people/lsxiao
 */

@Target(ElementType.METHOD)
@Retention(RetentionPolicy.CLASS)
public @interface SubscribeOn {
    SchedulerProvider.Tag value() default SchedulerProvider.Tag.IO;
}

File: core/src/main/java/com/lsxiao/apollo/core/annotations/SubscribeOn.kt
package com.lsxiao.apollo.core.annotations

import com.lsxiao.apollo.core.entity.SchedulerProvider


/**
 * write with Apollo
 * author:lsxiao
 * date:2017-04-22 02:21
 * github:https://github.com/lsxiao
 * zhihu:https://zhihu.com/people/lsxiao
 */

@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@kotlin.annotation.Retention
annotation class SubscribeOn(val value: SchedulerProvider.Tag = SchedulerProvider.Tag.IO)
